nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/MultimodeCommand.java

```diff
@@ -148,6 +158,88 @@ protected int executeCommand() throws CommandException, CommandValidationExcepti
         }
     }
 
+    private Prompter basicPrompter() throws IOException {
+        return new BufferedReaderPrompter(new BufferedReader(new InputStreamReader(System.in, Charset.defaultCharset())));
+    }
+
+    @Nonnull
+    private Prompter getPrompter() throws IOException {
+        if (DISABLE_JLINE) {
+            return basicPrompter();
+        }
+        try {
+            Terminal build = TerminalBuilder.builder()
+                    .system(true)
+                    .build();
+            SystemCompleter systemCompleter = new SystemCompleter();
+
+            habitat.getDescriptors(descriptor -> descriptor.getAdvertisedContracts().contains("com.sun.enterprise.admin.cli.CLICommand"))
+                    .forEach(activeDescriptor -> {
+                        String commandName = activeDescriptor.getName();
+                        if (commandName == null) {
+                            return;
+                        }
+                        if (commandName.startsWith("_")) {
+                            return;
+                        }
+                        try {
+                            habitat.reifyDescriptor(activeDescriptor);
+                            systemCompleter.add(commandName, getCompleter(commandName, activeDescriptor.getImplementationClass()));
+                        } catch (Exception ignored) {
+                        }
+                    });
+
+            systemCompleter.compile(Candidate::new);
+
+            LineReader lineReader = LineReaderBuilder.builder()
+                    .completer(new AggregateCompleter(systemCompleter, new RemoteCommandsCompleter()))
```

Comment:
Can you also add FileNameCompleter to autocomplete files, e.g. in deploy command?